### ISSUE TITLE
Video not playing after preRoll on iPhone

### ISSUE THREAD

**Issue by TeamBraha** [fluid-player/fluid-player#737]
**Describe the bug**
After preroll, fluid player tries to play the first video source, but fails since iPhone does not support webm. The first source is selected regardless of whether the device supports the file format or not.

If I remove vastOptions from the code everything works as it should and it plays the second video source (mp4) that is supported.

**Expected behavior**
First video source supported by browser/device is played after preroll.

   ```
 import {useEffect, useRef} from "react";
 import fluidPlayer from "fluid-player";

 export default function Video() {

    let self = useRef(null);
    let player = null;
    
    useEffect(() => {
      
      if (!player) {
        player = fluidPlayer(self.current,{
          debug: true,
          layoutControls: {
              fillToContainer:        true,
              allowTheatre:           false,
              playButtonShowing:      true,
              playPauseAnimation:     true,
              allowDownload:          false,
              playbackRateEnabled: true,
              controlBar: {
                autoHide:           true,
                autoHideTimeout:    3,
                animated:           true
            },
          },
          vastOptions: {
            adList: [
                {
                    roll: 'preRoll',
                    vastTag: '',
                },               
                {
                    roll: 'postRoll',
                    vastTag: '',
                }
            ]
        }
      });
      }
  });
  return (
    <>
        <video ref={self} >
          <source type="video/webm"  src="file.webm"/>
          <source type="video/mp4"    src="file.mp4" />
          Your browser does not support the video tag.
        </video>
    </>
  );
}
```
**Smartphone :**
 - Device: iPhone 12
 - Browser: safari & Chrome

**Affected version**
v3.16.0

**Comment by leroybm** [comment]
Hi @TeamBraha, I could reproduce the bug as described. I've raised an internal ticket for this.

**Comment by leroybm** [comment]
@TeamBraha I've tested this and found out this issue can happen regardless of Fluid Player. This is related to the way that Safari loads videos, if your video server is not setup correctly it can cause errors.

You can check a working video server in a demo page I've created, https://fp-218-prototype.glitch.me/, it seems that you need the correct type of response headers here. This demo page is just pure Javascript and HTML, you can go to https://fp-218-prototype.glitch.me/error.html to see the error happening outside of Fluid Player.

Let me know if changing the configuration solves this issue.

**Comment by leroybm** [comment]
@TeamBraha Have you been able to test this?

**Comment by leroybm** [comment]
Closing issue due to inactivity. @TeamBraha Please take a look at the previous comments and reopen the issue if it's still necessary.

**Comment by TeamBraha** [comment]
@leroybm I have now tested and found that this is not just a Safari problem, the same problem also occurs with Chrome on iPhone.

I configured my bucket that I use to serve the videos with the same response headers as in your example, but it made no difference.

Everything works as it should if I use a standard HTML5 video tag. The video also plays correctly if I remove vastOptions from Fluid Player, but then there is no point for me to use it.
